dracclient/resources/raid.py

```diff
@@ -98,15 +98,220 @@
 NO_FOREIGN_DRIVE = "STOR018"
 
 
+class RAIDAttribute(object):
+    """Generic RAID attribute class"""
+
+    def __init__(self, name, instance_id, current_value, pending_value,
+                 read_only, fqdd):
+        """Creates RAIDAttribute object
+
+        :param name: name of the RAID attribute
+        :param instance_id: InstanceID of the RAID attribute
+        :param current_value: current value of the RAID attribute
+        :param pending_value: pending value of the RAID attribute, reflecting
+                an unprocessed change (eg. config job not completed)
+        :param read_only: indicates whether this RAID attribute can be changed
+        :param fqdd: Fully Qualified Device Description of the RAID Attribute
+        """
+
+        self.name = name
+        self.instance_id = instance_id
+        self.current_value = current_value
+        self.pending_value = pending_value
+        self.read_only = read_only
+        self.fqdd = fqdd
+
+    def __eq__(self, other):
+        return self.__dict__ == other.__dict__
+
+    @classmethod
+    def parse(cls, namespace, raid_attr_xml):
+        """Parses XML and creates RAIDAttribute object"""
+
+        name = utils.get_wsman_resource_attr(
+            raid_attr_xml, namespace, 'AttributeName')
+        instance_id = utils.get_wsman_resource_attr(
+            raid_attr_xml, namespace, 'InstanceID')
+        current_value = utils.get_wsman_resource_attr(
+            raid_attr_xml, namespace, 'CurrentValue', nullable=True)
```

Comment:
I think there is an issue here.  I was looking at DCIM_RAIDEnumeration, and some of the attributes can have multiple CurrentValue's, like RAIDSupportedRAIDLevels.  I believe the code above will just take the first one found and set that as the current_value.  I think we're going to need to change current_value to current_values and make it an array.

Reply:
Wow, this is a much bigger problem.  Both CurrentValue and PendingValue are listed as an array in the 3.30.30.30 MOFs but we treat them as single values everywhere.

Reply:
Yes Agree Chris. I have checked some of the attributes in RAIDEnumeration , which is having multiple current value. I have updated code for that.

Reply:
> Wow, this is a much bigger problem. Both CurrentValue and PendingValue are listed as an array in the 3.30.30.30 MOFs but we treat them as single values everywhere.

I have checked the RAIDEnumeration attributes, we are having multiple current values and possible values but I have not found multiple pending values for any attribute. But need to test it once while setting the new raid setting when attribute value needed multiple current values to be set. Please correct me If I am getting wrong here.

Reply:
> I think there is an issue here. I was looking at DCIM_RAIDEnumeration, and some of the attributes can have multiple CurrentValue's, like RAIDSupportedRAIDLevels. I believe the code above will just take the first one found and set that as the current_value. I think we're going to need to change current_value to current_values and make it an array.

Yes Agree Chris. I have checked some of the attributes in RAIDEnumeration , which is having multiple current value. I have updated code for that. Doing one round of testing with new changes.